greynoise/plugin.spec.yaml

```diff
@@ -0,0 +1,509 @@
+      type: boolean
+      description: TOR
+      required: false
+  scan:
+    port:
+      title: "Port"
+      type: integer
+      description: "Port"
+      required: false
+    protocol:
+      title: "Protocol"
+      type: string
+      description: "Protocol"
+      required: false
+  raw_data:
+    hassh:
+      title: "HASSH"
+      type: "[]object"
+      description: "HASSH"
+      required: false
+    ja3:
+      title: "JA3"
+      type: "[]object"
+      description: "Ja3"
+      required: false
+    scan:
+      title: "Scan"
+      type: "[]scan"
+      description: "Scan"
+      required: false
+    web:
+      title: "Web"
+      type: object
+      description: "Web"
+      required: false
+  data:
+    ip:
+      title: IP Address
+      type: string
+      description: IP Address
+      required: false
+    first_seen:
+      title: GreyNoise First Seen
+      type: date
+      description: First Seen By GreyNoise
+      required: false
+    last_seen:
+      title: GreyNoise Last Seen
+      type: string
+      description: Last Seen By GreyNoise
+      required: false
+    seen:
+      title: GreyNoise Seen
+      type: boolean
+      description: Has this IP been Seen by GreyNoise
+      required: false
+    tags:
+      title: GreyNoise Tags
+      type: "[]string"
+      description: GreyNoise Tags Associated with IP
+      required: false
+    actor:
+      title: GreyNoise Actor
+      type: string
+      description: GreyNoise Actor Associated with IP
+      required: false
+    spoofable:
+      title: GreyNoise Spoofable
+      type: boolean
+      description: IP address may be spoofed
+      required: false
+    classification:
+      title: GreyNoise Classification
+      type: string
+      description: GreyNoise Classification
+      required: false
+    cve:
+      title: GreyNoise CVEs
+      type: "[]string"
+      description: CVEs associated with GreyNoise Tags
+      required: false
+    bot:
+      title: GreyNoise Bot
+      type: boolean
+      description: GreyNoise has identified this as a Bot
+      required: false
+    vpn:
+      title: GreyNoise VPN
+      type: boolean
+      description: GreyNoise has identified this as a VPN
+      required: false
+    vpn_service:
+      title: GreyNoise VPN Service
+      type: string
+      description: Name of VPN Service
+      required: false
+    metadata:
+      title: GreyNoise Metadata
+      type: metadata
+      description: GreyNoise IP Metadata
+      required: false
+    raw_data:
+      title: GreyNoise Raw Data
+      type: raw_data
+      description: GreyNoise IP Raw Data
+      required: false
+
+connection:
+  credentials:
+    title: GreyNoise API Key
+    description: API key from GreyNoise Account
+    type: credential_secret_key
+    required: true
+    example: abcdefghijklmnopqrstuvwxyz0123456789
+  server:
+    title: GreyNoise API URL
+    type: string
+    required: true
+    default: "https://api.greynoise.io"
```

Comment:
If this is not intended to change it could be worth omitting this input. Only caveat would be that if the URL changes the plugin will need to be updated but it would provide the user with only 1 configuration input of credentials.